### ISSUE TITLE
Update frequency

### ISSUE THREAD

**Issue by shashi007** [bradjasper/CryptocurrencySpreadsheetUtils#1]
Hi, Is there any way to improve the sheets update time, like every minute?

**Comment by bradjasper** [comment]
Hello, yes I believe there is a way to do this. At this line: https://github.com/bradjasper/CryptocurrencySpreadsheetUtils/blob/master/CryptocurrencySpreadsheetUtils.gs#L106

You can add another parameter as noted here https://developers.google.com/apps-script/reference/cache/cache#putAll(Object,Integer) which is the timeout parameter.

Right now it is the default at 10 minutes. If you'd like to submit a pull request for this as a configurable option at the top of the script I'd accept it.

Cheers!

**Comment by shashi007** [comment]
Hi, Thanks for the reply. I tried that way but sheet didn't update.
` cache.putAll(cachedCoins, 60);`

**Comment by bradjasper** [comment]
Was it previously cached? I don't think the sheet will update automatically, but should update if you refresh the page.

**Comment by bradjasper** [comment]
I researched this a bit more and found an inelegant solution—I'm still hoping someone finds an alternative.

It seems Google Sheets caches the function call and arguments itself, so the internal cache may not even matter. The obvious way to bust this would be pass a random value in, but Google Sheets disallows NOW() and RAND()—two good candidates.

I found another solution that allows you to populate a random cell from the menubar, but it requires a hacky authorization and still requires manual updating.

If you want to try it out, add this piece of code:

```
function getCoinPriceCB(symbol, cacheBust) {
  return getCoinPrice(symbol);
}
```

Now you can do `getCoinPriceCB("SYM", $A$1)`

Now in $A$1 you can put a random value, or just increment it (1, 2, 3, 4) and prices will update every time. I'll see if I can figure out a way to make the menubar not request authorization.

**Comment by bradjasper** [comment]
This was recently fixed in v0.2—it still requires a manual operation, but you can add a refresh button to make it 1-click: https://github.com/bradjasper/CryptocurrencySpreadsheetUtils#refresh

**Comment by jeromedalbert** [comment]
If anyone finds this interesting, I personally don't use in-cell formula, and use a custom function in the script editor. I use the code below but most people can do something much simpler:

```
function refreshRates() {
  var sheets = SpreadsheetApp.getActiveSpreadsheet();
  var rates = sheets.getSheetByName('Rates');
  var cryptoSymbols = sheets.getRange('Portfolio!B6:B1000').getValues();

  for (var i = 0; i < cryptoSymbols.length; i++) {
    var symbol = cryptoSymbols[i][0];
    if (!symbol) return;
    
    rates.getRange(i+2, 1).setValue(symbol);
    rates.getRange(i+2, 2).setValue(getCoinPrice(symbol));
  }
}
```

Then in the script editor, click `Edit > Current project's triggers` and set your function to execute at any interval you want! I personally set 5 minutes. The advantage of this method is that it's much more reliable: the Google servers will update your spreadsheet independently from you. Very powerful!

Plus, with this method I can check my rates with the Google Sheets iphone app, and it will work without any lag or loading time.
